mobile-app/app/screens/AppNavigator/screens/Dex/components/PoolPairCards/PoolPairCards.tsx

```diff
@@ -0,0 +1,426 @@
+import NumberFormat from 'react-number-format'
+import { ActiveUSDValue } from '@screens/AppNavigator/screens/Loans/VaultDetail/components/ActiveUSDValue'
+import { ButtonGroup } from '../ButtonGroup'
+import { useSelector } from 'react-redux'
+import { RootState } from '@store'
+import { TotalValueLocked } from '../TotalValueLocked'
+interface DexItem<T> {
+  type: 'your' | 'available'
+  data: T
+}
+
+export enum ButtonGroupTabKey {
+  AllPairs = 'All_PAIRS',
+  DFIPairs = 'DFI_PAIRS',
+  DUSDPairs = 'DUSD_PAIRS'
+}
+
+interface PoolPairCardProps {
+  availablePairs: Array<DexItem<PoolPairData>>
+  yourPairs: Array<DexItem<WalletToken>>
+  onAdd: (data: PoolPairData) => void
+  onRemove: (data: PoolPairData) => void
+  onSwap: (data: PoolPairData) => void
+  type: 'your' | 'available'
+  setIsSearching: (isSearching: boolean) => void
+  searchString: string
+  buttonGroupOptions?: {
+    onButtonGroupPress: (key: ButtonGroupTabKey) => void
+    activeButtonGroup: string
+    setActiveButtonGroup: (key: ButtonGroupTabKey) => void
+  }
+  showSearchInput?: boolean
+}
+
+export function PoolPairCards ({
+  availablePairs,
+  onAdd,
+  onRemove,
+  onSwap,
+  type,
+  searchString,
+  setIsSearching,
+  yourPairs,
+  buttonGroupOptions,
+  showSearchInput
+}: PoolPairCardProps): JSX.Element {
+  const { poolpairs: pairs } = useSelector(
+    (state: RootState) => state.wallet
+  )
+  const { isFavouritePoolpair, setFavouritePoolpair } = useFavouritePoolpairs()
+  const sortedPairs = sortPoolpairsByFavourite(
+    availablePairs,
+    isFavouritePoolpair
+  )
+  const { getTokenPrice, calculatePriceRates } = useTokenPrice()
+  const [expandedCardIds, setExpandedCardIds] = useState<string[]>([])
+
+  const [filteredYourPairs, setFilteredYourPairs] =
+    useState<Array<DexItem<WalletToken>>>(yourPairs)
+  const debouncedSearchTerm = useDebounce(searchString, 2000)
+  const { tvl } = useSelector((state: RootState) => state.block)
+  const buttonGroup = [
+    {
+      id: ButtonGroupTabKey.AllPairs,
+      label: translate('screens/DexScreen', 'All pairs'),
+      handleOnPress: () => onButtonGroupChange(ButtonGroupTabKey.AllPairs)
+    },
+    {
+      id: ButtonGroupTabKey.DFIPairs,
+      label: translate('screens/DexScreen', 'DFI pairs'),
+      handleOnPress: () => onButtonGroupChange(ButtonGroupTabKey.DFIPairs)
+    },
+    {
+      id: ButtonGroupTabKey.DUSDPairs,
+      label: translate('screens/DexScreen', 'DUSD pairs'),
+      handleOnPress: () => onButtonGroupChange(ButtonGroupTabKey.DUSDPairs)
+    }
+  ]
+  const onButtonGroupChange = (buttonGroupTabKey: ButtonGroupTabKey): void => {
+    if (buttonGroupOptions !== undefined) {
+      buttonGroupOptions.setActiveButtonGroup(buttonGroupTabKey)
+      buttonGroupOptions.onButtonGroupPress(buttonGroupTabKey)
+    }
+  }
+
+  useEffect(() => {
+    setIsSearching(false)
+    setFilteredYourPairs(
+      yourPairs.filter((pair) =>
```

Comment:
I think caching this can improve a bit of performance

Reply:
I cant test on android since i have input issue, but no performance observed on my ios device. Maybe @pierregee can check